tests/framework/binding.h

```diff
@@ -220,22 +220,29 @@ class Device : public internal::Handle<VkDevice> {
     // vkGetDeviceProcAddr()
     PFN_vkVoidFunction get_proc(const char *name) const { return vk::GetDeviceProcAddr(handle(), name); }
 
-    // vkGetDeviceQueue()
-    const std::vector<Queue *> &graphics_queues() const { return queues_[GRAPHICS]; }
-    const std::vector<Queue *> &compute_queues() { return queues_[COMPUTE]; }
-    const std::vector<Queue *> &dma_queues() { return queues_[DMA]; }
-    const std::vector<Queue *> &sparse_queues() { return queues_[SPARSE]; }
+    const std::vector<Queue *> &QueuesWithGraphicsCapability() const { return queues_[GRAPHICS]; }
+    const std::vector<Queue *> &QueuesWithComputeCapability() const { return queues_[COMPUTE]; }
+    const std::vector<Queue *> &QueuesWithTransferCapability() const { return queues_[TRANSFER]; }
+    const std::vector<Queue *> &QueuesWithSparseCapability() const { return queues_[SPARSE]; }
 
     typedef std::vector<std::unique_ptr<Queue>> QueueFamilyQueues;
     typedef std::vector<QueueFamilyQueues> QueueFamilies;
     const QueueFamilyQueues &queue_family_queues(uint32_t queue_family) const;
 
-    // Find a queue family with and without desired capabilities
-    std::optional<uint32_t> QueueFamilyMatching(VkQueueFlags with, VkQueueFlags without, bool all_bits = true);
-    std::optional<uint32_t> QueueFamilyWithoutCapabilities(VkQueueFlags capabilities) {
-        // an all_bits match with 0 matches all
-        return QueueFamilyMatching(VkQueueFlags(0), capabilities, true /* all_bits with */);
-    }
+    // Queue family that has "with" capabilities and optionally without "without" capabilities.
+    std::optional<uint32_t> QueueFamily(VkQueueFlags with, VkQueueFlags without = 0) const;
+
+    // Queue family that does not have "without" capabilities
+    std::optional<uint32_t> QueueFamilyWithoutCapabilities(VkQueueFlags without);
+
+    // Dedicated compute queue family: has compute but no graphics
+    std::optional<uint32_t> ComputeQueueFamily() const;
```

Comment:
```suggestion
    std::optional<uint32_t> ComputeOnlyQueueFamily() const;
```

might be easier to tell right away it isn't "any queue that has compute"

Reply:
Updated

Reply:
I was thinking about "only" during development. I agree it gives proper intuition when reading code.  Was not sure if someone wants to use "only" queue to check for other caps and whether "only" can be a bit misleading because "only" queues can also have other caps. We probably can introduce more helpers if needed.

One option for vvl/driver apps is this naming: `Dedicated{Compute/Transfer}Queue`

Another option is `Async{Compute/Transfer}Queue`, but that's might be more specific for graphics apps, which usually use "only" queues for async operations. When async is not needed, they likely will run, for example, compute commands on the graphics queue, even if there is a compute-only queue.